google/cloud/storage/internal/policy_document_request.cc

```diff
@@ -275,11 +275,9 @@ std::string PolicyDocumentV4Request::Url() const {
     return scheme_ + "://" + *bucket_bound_domain_ + "/";
   }
   if (virtual_host_name_) {
-    return scheme_ + "://" + policy_document().bucket +
-           ".storage.googleapis.com/";
+    return scheme_ + "://" + policy_document().bucket + "." + host_ + "/";
```

Comment:
Please change the unit tests to validate the new functionality.

Reply:
Added new unit tests in `policy_document_request_test`